.github/workflows/build-binaries.yaml

```diff
@@ -37,6 +39,12 @@ jobs:
       - name: Set up Poetry
         uses: ./.github/actions/setup-poetry
 
+      - name: Setup poetry cache
```

Comment:
Any specific reason for putting this back cause i moved it into the setup poetry action in the previous pr